internal/devbox/packages.go

```diff
@@ -43,6 +44,48 @@ const StateOutOfDateMessage = "Your devbox environment may be out of date. Run %
 // packages.go has functions for adding, removing and getting info about nix
 // packages
 
+type UpdateVersion struct {
+	Current string
+	Latest  string
+}
+
+// Outdated returns a map of package names to their available latest version.
+func (d *Devbox) Outdated(ctx context.Context) (map[string]UpdateVersion, error) {
+	ctx, task := trace.NewTask(ctx, "devboxOutdated")
+	defer task.End()
+
+	outdatedPackages := map[string]UpdateVersion{}
+
+	for _, pkg := range d.AllPackages() {
+		if strings.HasSuffix(pkg.Versioned(), "latest") {
+			continue
+		}
+
+		result, err := searcher.Client().Search(ctx, pkg.CanonicalName())
+		if err != nil {
+			return nil, err
+		}
```

Comment:
@gcurtis does this reliably give us the latest version of the package?

For example, when I run it with `go` I get:
```
 devbox ls --outdated
The following packages can be updated:
 * go                             1.23.4 -> 1.24rc2
```

which doesn't seem to be what we want

Reply:
I guess this is correct, is the latest version available in the store :)

Reply:
But its a release-candidate. In `devbox update` we wouldn't actually update you to that version if you had `go@latest`